DancingGoat/Helpers/DeliveryClientFactory.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+
+using DancingGoat.Models;
+using DancingGoat.Utils;
+
+using KenticoCloud.Delivery;
+
+namespace DancingGoat.Helpers
+{
+    public class DeliveryClientFactory
+    {
+        public static DeliveryClient CreateDeliveryClient()
+        {
+            // Use the provider to get environment variables.
+            var provider = new ConfigurationManagerProvider();
+
+            // Build DeliveryOptions with default or explicit values.
+            var options = provider.GetDeliveryOptions();
+
+            options.ProjectId = GetProjectId();
+
+            var clientInstance = new DeliveryClient(options);
+            clientInstance.CodeFirstModelProvider.TypeProvider = new CustomTypeProvider();
+            clientInstance.ContentLinkUrlResolver = new CustomContentLinkUrlResolver();
+            return clientInstance;
+        }
+
+        public static string GetProjectId()
+        {
+            var requestUrl = System.Web.HttpContext.Current.Request.Url;
+            var subdomain = requestUrl.GetSubdomain();
+            var decodedGuid = GuidUtils.FromShortString(subdomain);
+
+            if (decodedGuid == null)

```

Comment:
Guid can't be null, therefore this will never be true. It will either be Guid.Empty, or the parsing (ParseExact) will throw an exception on its own